server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/JmapRuleDTO.java

```diff
@@ -151,8 +151,6 @@ public JmapRuleDTO(@JsonProperty("id") String id,
         this.name = name;
         this.conditionDTO = conditionDTO;
         this.actionDTO = actionDTO;
-        Preconditions.checkNotNull(id);
```

Comment:
That's unfortunate, by this we allow the "id" field of the JmapRuleDTO object to be null.

We should reject invalid values as early as possible. Here I would advocate the use of Preconditions for every field...